kernel/nic_queues/src/lib.rs

```diff
@@ -43,21 +77,106 @@ pub struct RxQueue<T: RxDescriptor> {
     pub received_frames: VecDeque<ReceivedFrame>,
     /// The cpu which this queue is mapped to. 
     /// This in itself doesn't guarantee anything, but we use this value when setting the cpu id for interrupts and DCA.
-    pub cpu_id: u8,
+    pub cpu_id: Option<u8>,
+    /// Pool where `ReceiveBuffer`s are stored.
+    pub rx_buffer_pool: &'static mpmc::Queue<ReceiveBuffer>,
+    /// The filter id for the physical NIC filter that is set for this queue
+    pub filter_num: Option<u8>
 }
 
+impl<S: RxQueueRegisters, T: RxDescriptor> RxQueue<S,T> {
+    /// Retrieves the ethernet frames from one queue
+    pub fn remove_frames_from_queue(&mut self) -> Result<(), &'static str> {
```

Comment:
I can guess here, but the behavior of this fn isn't 100% clear from the docs and name. I _assume_ it removes _all existing_ frames that are "ready" (ready means what?) from this queue, but I'm not sure. Docs/name improvements could help.